more documentation for non-standard roles (#999)
tidymodels/recipes: man/rmd/bake_dependent_roles.Rmd

## Effects of non-standard roles

> Recipes steps can keep column(s) of your data set that should not be treated as outcomes or predictors. For example, a unique identifier column or some other ancillary data that can be used to troubleshoot issues during model development. 

Reviewer: ```suggestion
Recipes can label and retain column(s) of your data set that should not be treated as outcomes or predictors. For example, a unique identifier column or some other ancillary data that can be used to troubleshoot issues during model development.
```

> Recipes can label and retain column(s) of your data set that should not be treated as outcomes or predictors. For example, a unique identifier column or some other ancillary data that can be used to troubleshoot issues during model development.

Reviewer 2: ```suggestion
Recipes can label and retain column(s) of your data set that should not be treated as outcomes or predictors. A unique identifier column or some other ancillary data could be used to troubleshoot issues during model development but may not be either an outcome or predictor.
```

> As an example, you might have a date column that is transformed into features using [step_date()] and.or [step_holiday()]. 

Reviewer: ```suggestion
As an example, you might have a date column that is transformed into features using [step_date()] or [step_holiday()]. 
```

```{r, results = "hide", message = FALSE, warning = FALSE}

>   library(tidymodels)

Reviewer: It looks like all of the code in your code chunks are indented, was that on purpose?

Author: it is!

  tidymodels_prefer()

  data(Chicago, package = "modeldata")
  chi_rec <- recipe(ridership ~ ., data = Chicago) %>% 
    # Since `date` is symbolically on the right-hand side of the formula, it
    # is treated as a predictor (i.e., has a role of "predictor")
    step_holiday(date) %>% 

>     step_date(date)

Reviewer 2: Can we use something here that is not `date`? I think this gets really muddy because it is sort of a predictor and you are using it to create predictors, and we have better ways to not keep the original variable around. I think `tate_text` would be a better option for these docs.

Reviewer 2: I've spent a bit more time on this, and I do think we can do better than the `date` example for being clear about how to use these other roles.

Author: This is exactly the application that has tried people up (us included). It seems most relevant. I can add another example though.

```

At this pont, `date` is kept in the dataset as-is, and will be treated as a predictor (and will probably be converted to an integer column by the modeling function). 

> One option is to remove date since it is not directly used as a model term. [step_rm()] is a good way to accomplish this. 

Reviewer 3: ```suggestion
One option is to remove `date` since it is not directly used as a model term. [step_rm()] is a good way to accomplish this. 
```

```{r}
  chi_rec_rm <- chi_rec %>% step_rm(date)
```

Another approach, especially helpful during model development, is to change the role of the column to some other value via [update_role()]. 

```{r}
  chi_rec_role <- chi_rec %>% update_role(date, new_role = "date")
```

> In this case, the column is still available in the data and will be carried through the analysis (e.g. during resampling). This can be helpful since, it there is a poor prediction, this column may be helpful in diagnosing the problem. 

Reviewer 3: ```suggestion
In this case, the column is still available in the data and will be carried through the analysis (e.g. during resampling). This can be helpful since, if there is a poor prediction, this column may be helpful in diagnosing the problem. 
```

The downside to chaining the role of the column is that it is needed at prediction-time. This can impact development tools (such as tuning functions) and would also be required to be in the data being predicted once the model is deployed. 

> If using a non-standard role for a column, it highly recommended that you use the recipe within a workflow and specify `bake_dependent_roles` as described in [hardhat::default_recipe_blueprint()]. This will allow tuning and prediction functions to run without error. 

Reviewer 3: ```suggestion
If using a non-standard role for a column, it is highly recommended that you use the recipe within a workflow and specify `bake_dependent_roles` as described in [hardhat::default_recipe_blueprint()]. This will allow tuning and prediction functions to run without error. 
```

> If using a non-standard role for a column, it is highly recommended that you use the recipe within a workflow and specify `bake_dependent_roles` as described in [hardhat::default_recipe_blueprint()]. This will allow tuning and prediction functions to run without error. 

Reviewer 2: If we're putting this in a workflow already in these docs, it would be very helpful (probably more helpful than the `bake_dependant_roles` demo) to show that you can `augment()` your fitted workflow using variables that aren't outcomes or predictors.
